cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ExistingDataprocProvisioner.java

```diff
@@ -145,6 +145,10 @@ public Cluster getClusterDetail(ProvisionerContext context, Cluster cluster) {
 
   @Override
   public PollingStrategy getPollingStrategy(ProvisionerContext context, Cluster cluster) {
+    if (cluster.getStatus() == ClusterStatus.DELETING) {
+      DataprocConf conf = DataprocConf.create(createContextProperties(context));
+      return PollingStrategies.fixedInterval(conf.getPollInterval(), TimeUnit.SECONDS);
+    }
     return PollingStrategies.fixedInterval(0, TimeUnit.SECONDS);
```

Comment:
Why do we want 0 at all? We should have some minimum delay, even if few hundred millis.

Reply:
As discussed offline, reversed the if. In case cluster status is CREATING for existing dataproc clusters poll interval will be 0 sec. Otherwise, we use getPollInterval value.